selfhost/typechecker.jakt

```diff
@@ -8972,6 +8972,10 @@ struct Typechecker {
                     let array_slice_struct_id = .find_struct_in_prelude("ArraySlice")
                     let dictionary_struct_id = .find_struct_in_prelude("Dictionary")
 
+                    if not id.equals(array_struct_id) and not id.equals(array_slice_struct_id) and not id.equals(dictionary_struct_id) {
+                        .error(format("Type ‘{}’ does not implement the indexing operation", .type_name(checked_base.type())), span)
```

Comment:
it's impossible to "implement" the indexing operation (at least atm), so maybe:
```suggestion
                        .error(format("Type ‘{}’ cannot be indexed by []", .type_name(checked_base.type())), span)
```

Reply:
Now it's possible to implement it

Reply:
well...I suppose it is :)